snapcraft_legacy/cli/lifecycle.py

```diff
@@ -103,6 +104,15 @@ def _execute(  # noqa: C901
 
     if build_provider in ["host", "managed-host"]:
         project_config = project_loader.load_config(project)
+
+        # validate experimental plugins
+        plugins = [part.plugin for part in project_config.parts.all_parts]
+        if not kwargs.get("enable_experimental_plugins") and any(isinstance(plugin, KernelPlugin) for plugin in plugins):
```

Comment:
This is slightly simpler and less extensible than Claudio's core22 implementation.
I think this is OK because it is unlikely that we will add more core20 experimental plugins in the future.